Setup rabbitmq-messagebus: selftest with initial 10 tests

duongtdn opened the pull request:
an initial 10 tselftest for rabbitmq-messagebus are created.
adapted testtool and strategy from json-preprocessor for rabbitmq-messagebus.
closing #3

duongtdn committed:
test: complete initial test setup

- update config and runner for testfiles support
- add initial dummy tests which will be replaced later

duongtdn committed:
test: Add async EventBusClient tests and utilities

- Updated component_test.py for async EventBusClient setup and cleanup.
- Implenmented EBC_0001 and EBC_0002 tests for message flow.
- Added config.jsonp for EventBusClient configuration.
- Added SimpleTestMessage for testing.

duongtdn committed:
test: add EBC_0003 test case for multiple subscribers

duongtdn committed:
test: add EBC_0004 test case for multiple publishers

duongtdn committed:
refactor: move EventBusClient creation to test function for flexible config testing

- Change test category for created tests to TopicExchangeHandler
- Add additional test EBC_00005

duongtdn committed:
test: add test EBC_0006

duongtdn committed:
test: add test EBC_0007

duongtdn committed:
test: add test EBC_0008

duongtdn committed:
test: add test EBC_0009

duongtdn committed:
test: add test EBC_0010

duongtdn committed:
test: add helper script to run test on linux

duongtdn committed:
chore: add docker for linux environment testing

duongtdn committed:
test: add generated pytest files

duongtdn commented:
Waiting PR #14 to be merged to add further tests.
Note: CI test may failed untill PR #14 is merged to solve import issues.

duongtdn committed:
doc: add test list files

HolQue reviewed (commented):
Hi Duong,

very good! Selftest runs properly on my computer without any issues.

test-fullautomation commented on test/testfiles/EBC_0001.py:
```diff
@@ -0,0 +1,84 @@
+        await oEventBusClient.send(routing_key, test_message)
+
+        # Wait for message to be processed
+        await asyncio.sleep(0.5)
```
Hi Duong, 
this fix waiting times can cause sporadically failing tests if because of whatever reason the time was not long enough. Better would be here a polling of received_messages with a timeout in case of error (e.g. after 5 seconds no result).

Same in line 63. Would be better to poll readiness somehow.

This finding applies for all tests.
Can you please check with Cuong if there is a better way. Maybe he can also provide an API which allows syncing.  This is a general issue also in prodcution use of the rabbitemq messagebus client.

Thank you,
 Thomas

duongtdn replied:
Hi Thomas,

Thanks for your findings.

I have added 2 commits. They implement some test polling helper utilities, and fix 10 tests to use these polling helpers instead of fixed wait time.

Regards,
Duong

test-fullautomation reviewed (changes requested):
Hi Duong,

thank you for your pull request.
Please check my findings.

Thank you,
 Thomas

duongtdn committed:
enhance(test): polling utilities for EventBus tests

- Add polling_utils.py module with timeout and retry logic
- Replace hardcoded asyncio.sleep() calls with polling
- Update EBC_0001-EBC_0010 test cases to use new polling mechanisms

duongtdn committed:
enhance(test): add wait_for_client_connected utility

- Add wait_for_client_connected utility function to polling_utils.py
- Update all test files (EBC_0001-EBC_0010) to wait for client connection before sending messages

test-fullautomation reviewed (approved):
Hi Duong,

nice code now!

Than you,
 Thomas
